glue/core/data_factories/tests/test_numpy.py

```diff
@@ -20,6 +20,31 @@ def test_npy_load(tmpdir):
         assert_array_equal(data['ra'], data2['ra'])
         assert_array_equal(data['dec'], data2['dec'])
 
+def test_unstruc_npy_load(tmpdir):
+    data = np.array([[152.2352, -21.513], [21.412, 35.1341]], dtype='f8')
+
+    with open(tmpdir.join('test.npy').strpath, 'wb') as f:
+        np.save(f, data)
+        f.seek(0)
+
+        data2 = df.load_data(f.name)
+        assert_array_equal(data, data2['array'])
```

Comment:
You shouldn't need the ``seek(0)`` - instead, move the ``data2=`` and the following line out of the ``with`` statement (up one indent level) and this will ensure the file is closed. I would also set:

```python
filename = tmpdir.join('test.npy').strpath
```

before the ``with`` statement and use that in the ``with`` statement and for ``load_data``